src/ui/widgets/MenuButton/menuButton.tsx

```diff
@@ -21,129 +25,169 @@ import {
 import { FileContext } from "../../../misc/fileContext";
 import { Border } from "../../../types/border";
 import { Color } from "../../../types/color";
+import { MenuItem, Select, SelectChangeEvent } from "@mui/material";
+import { diamondTheme } from "../../../diamondTheme";
+import { Font } from "../../../types";
 
 export interface MenuButtonProps {
   connected: boolean;
   onChange: (action: WidgetAction) => void;
   pvName?: string;
   value?: DType;
-  readonly: boolean;
   actionsFromPv?: boolean;
   itemsFromPv?: boolean;
   label?: string;
   actions?: WidgetActions;
   foregroundColor?: Color;
   backgroundColor?: Color;
   border?: Border;
+  font?: Font;
+  items?: string[];
+  enabled?: boolean;
 }
 
+const MenuButtonComponentProps = {
+  pvName: StringPropOpt,
+  foregroundColor: ColorPropOpt,
+  backgroundColor: ColorPropOpt,
+  font: FontPropOpt,
+  enabled: BoolPropOpt,
+  value: DTypePropOpt,
+  actions: ActionsPropType,
+  connected: BoolPropOpt,
+  onChange: FuncPropOpt,
+  // opi specific prop
+  actionsFromPv: BoolPropOpt,
+  label: StringPropOpt,
+  // bob specific prop
+  items: StringArrayPropOpt,
+  itemsFromPv: BoolPropOpt
+};
+
```

Comment:
`value`, `pvName` and `connected` (and I think `actions` as well) don't need to be specified here. `MenuButtonComponentProps` is only used when we combine it with `PVWidgetPropType` (which already contains these properties) to create `MenuButtonWidgetProps`. This also means you don't need to create the additional `DTypePropOpt` type in propTypes.ts